src/php/lib/Grpc/AbstractCall.php

```diff
@@ -141,12 +134,7 @@ protected function _deserializeResponse($value)
         if (is_array($this->deserialize)) {
             list($className, $deserializeFunc) = $this->deserialize;
             $obj = new $className();
-            if (method_exists($obj, $deserializeFunc)) {
-                $obj->$deserializeFunc($value);
-            } else {
-                $obj->mergeFromString($value);
-            }
-
+            $obj->mergeFromString($value);
             return $obj;
         }
 
```

Comment:
Can you also remove the next 2 lines 
```
        // Protobuf-PHP implementation
        return call_user_func($this->deserialize, $value);
```

Then this is parallel to the work you did up in `_serializeMessage`.

Then you can also remove the `if (is_array($this->deserialize)) {` wrap.

So the entire `_deserializeResponse` function should be something like

```
    protected function _deserializeResponse($value)
    {
        if ($value === null) {
            return;
        }

        list($className, $deserializeFunc) = $this->deserialize;
        $obj = new $className();
        $obj->mergeFromString($value);
        return $obj;
    }
```

Reply:
got it